open-edge-platform/training_extensions
# Add ViT-B for visual prompting

### Summary


This PR includes:
- [x] ViT-B for visual prompting and zero-shot visual prompting
    - [x] Empty prediction in `PromptGetter` (zero-shot)
    - [x] Stack more indices than #masks in `__inspect_overlapping_areas` (zero-shot)
- [x] Improve managing `reference_info` (zero-shot)

Comparing with v1.5, fine-tuning results on wgisd dataset to check reproducibility are shown below:
version | best epoch | best mDice
-- | -- | --
v1.5 | 11 | 93.36
v2.0 | 81 | 93.33


TODOs :
- set regression test in the next PR
- update recipe
    - Performance ranges of both versions seems the same, but I'm not sure why v2.0's early stopping worked late.
![image](https://github.com/openvinotoolkit/training_extensions/assets/97221135/a2133e00-f4f2-4230-ad86-3ee2d2fbacde)

### How to test



### Checklist




### License

      Feel free to contact the maintainers if that's a concern.

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```
